src/main/java/nl/vpro/jetty/MagnoliaWebAppClassLoader.java

```diff
@@ -155,60 +155,79 @@ protected  void watch() {
-                        for (File d : dirs) {
-                            if (context.toAbsolutePath().startsWith(d.getAbsolutePath())) {
-                                LOG.info("Found new directory in {} in {}, watching too", context, d);
-                                try {
-                                    context.register(watchService, ENTRY_CREATE);
-                                } catch (IOException e) {
-                                    LOG.error(e.getMessage());
+                            if (! found){
+                                LOG.warn("Could not find anytihng to match {}", event);
+                            }
+                        } else if (Files.isDirectory(context)) {
+                            for (File d : dirs) {
+                                if (context.toAbsolutePath().startsWith(d.getAbsolutePath())) {
+                                    LOG.info("Found new directory {} in {} in {}, watching too", event.kind(), context, d);
+                                    try {
+                                        WatchKey newKey = context.register(watchService, ENTRY_CREATE);
+                                        keys.put(newKey, context);
+                                        LOG.info("Now watching {} directories for new files", keys.size());
+                                        touch(d); // new directory may contain files
+                                    } catch (IOException e) {
+                                        LOG.error(e.getMessage());
+                                    }
+                                    break;
                                 }
-                                break;
                             }
                         }
-                    }
 
+                    }
+                } finally {
+                    valid = key.reset(); // IMPORTANT: The key must be reset after processed
                 }
-                valid = key.reset(); // IMPORTANT: The key must be reset after processed
             } catch (InterruptedException e) {
                 LOG.info("Interrupted");
                 return;
             }
         }
     }
 
+    protected void touch(File d) {
+        long lastModified = System.currentTimeMillis();
+        if (d.lastModified() < lastModified) {
+            boolean success = d.setLastModified(lastModified);
+            if (!success) {
+                LOG.warn("Could not set timestamp of {}", d);
+            }
+        }
+    }
+
     protected Map<WatchKey, Path>  registerWatchers()  {
         final Map<WatchKey, Path> keys = new HashMap<>();
 
          for (File watchedDirectory : dirs) {
             Path path = Paths.get(watchedDirectory.toString());
             try {
-                Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
+                Files.walkFileTree(path, EnumSet.of(FileVisitOption.FOLLOW_LINKS), Integer.MAX_VALUE, new SimpleFileVisitor<Path>() {
```

Comment:
https://stackoverflow.com/questions/42158761/java-nio-iterate-files-in-symbolic-link dit was het trucje dus?

Reply:
Nou ja truukje. Ik had t gewoon in de source/javadoc gezien  :)